### PR TITLE
page-form - boolean field

### PR DESCRIPTION
**Pull request by collin**

### CODE REVIEW

File: app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-boolean/dynamic-content-form-field-boolean.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<dynamic-content-form-field-input-label
+    name="{{ name }}"
+    schema="schema">
+</dynamic-content-form-field-input-label>
+<input
+    name="{{ name }}"
+    ng-model="ngModel"
+    type="checkbox"
+    ng-required="schema.required">
+<dynamic-content-form-field-input-errors
```
Comment (collin) [inline app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-boolean/dynamic-content-form-field-boolean.html]: @kand Do we need the field errors here?

Boolean should be either true or false. Is there a case for a `required` error here?

Will there be a readonly case for this input?

**Comment by collin** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;uqQTnU7S&#x2F;42-1-boolean"> (1) boolean</a>

**Comment by collin** [comment]
@kand and this one too

File: app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-boolean/dynamic-content-form-field-boolean.html (hunk 1 above)
Comment (kand) [inline app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-boolean/dynamic-content-form-field-boolean.html]: The [infographics_spec](https://gist.github.com/benghaziboy/c583e1e56de2d851403e6553ec2f7727) I have has boolean fields, but all with `read_only` and `required` as `false`. I think we might as well implement these validations for this field.

File: app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-object/dynamic-content-form-field-object.tests.js [hunk 2]
```diff
@@ -147,6 +148,19 @@ describe('Directive: dynamicContentFormFieldObject', function () {
     expect(html.find('dynamic-content-form-field-text').length).to.equal(1);
   });
 
+  it('should render a boolean field when given a field with type text', function () {
```
Comment (kand) [inline app/components/dynamic-content/dynamic-content-form/dynamic-content-form-field/dynamic-content-form-field-object/dynamic-content-form-field-object.tests.js]: The text for this test should read `"... with type boolean"`.

**Comment by kand** [comment]
Aside from my comments, this looks good 👍

**Pull request merged**